src/crypto/chacha20.cpp

```diff
@@ -23,6 +24,10 @@ static const unsigned char tau[] = "expand 16-byte k";
 
 void ChaCha20::SetKey(const unsigned char* k, size_t keylen)
 {
+    if (keylen != 16 && keylen != 32) {
+        throw std::invalid_argument("ChaCha20 key size must be 16 or 32");
```

Comment:
Wouldn't it be better to make this an assert? This would allow to compile the crypto code with `noexcept`

Reply:
I too think an assert would be more appropriate here.  It's unclear how an exception will be handled. Also the caller is responsible that a valid key is passed. When this doesn't happen it is a coding bug, it makes more sense to immediately terminate the process.

Reply:
Replaced with assert.